### ISSUE TITLE
[JENKINS-35815] Optimise JenkinsJSExtensions.getJenkinsJSExtensionData to cache extensions

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#2783]
<p>JenkinsJSExtensions.getJenkinsJSExtensionData is serializing a Map on every single call at the moment, which is not a very scalable solution as number of plugins and extensions grow. </p>

<p>This could be cached in some fashion</p>

<p>cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly" class="user-hover" rel="tfennelly">tfennelly</a></p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35815" target="_blank">Optimise JenkinsJSExtensions.getJenkinsJSExtensionData to cache extensions</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: blueocean-imported, technical-debt
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2016-09-05T01:38:35+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>JenkinsJSExtensions.getJenkinsJSExtensionData is serializing a Map on every single call at the moment, which is not a very scalable solution as number of plugins and extensions grow. </p>

<p>This could be cached in some fashion</p>

<p>cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly" class="user-hover" rel="tfennelly">tfennelly</a></p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35815?focusedId=261283&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261283">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Note that every call is only once per page load - so timing is important for the main page load. Putting this in the main html returned for the page would also be a better idea, as this ideally wouldn't be a REST call. </p>
</pre>
</details>

</ul>
</details>

<p>Note that every call is only once per page load - so timing is important for the main page load. Putting this in the main html returned for the page would also be a better idea, as this ideally wouldn't be a REST call. </p>
